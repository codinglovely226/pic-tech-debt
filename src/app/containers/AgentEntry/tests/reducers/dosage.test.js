/* eslint-disable */

import dosageReducer from '../../reducers/dosage';

import {
  addAgent,
  updateAgentQuantity,
  updateQuickViewAgentQuantity,
  updateQuickViewAgentQuantityUnits,
  updateQuickViewAgentQuantityUnknown,
} from '../../actions/case';

describe('dosageReducer', () => {
  it('returns the initial state', () => {
    expect(dosageReducer(undefined, {})).toMatchSnapshot();
  });
  it('add agent', () => {
    expect(dosageReducer({}, addAgent(2, 'Test Agent', '1', 'mg', false, '2018-04-14 14:25:00', '1', 'hour', false, 'Test Exposure Type', 'Ingested', '123456'))).toMatchSnapshot();
  });
  it('update agent quantity', () => {
    expect(dosageReducer({}, updateAgentQuantity(1, 1, '1', '2', 'mg', 'g', false, false))).toMatchSnapshot();
  });
  it('update quick view agent quantity', () => {
    expect(dosageReducer({}, updateQuickViewAgentQuantity(10))).toMatchSnapshot();
  });
  it('update quick view agent quantity units', () => {
    expect(dosageReducer({}, updateQuickViewAgentQuantityUnits('kg'))).toMatchSnapshot();
  });
  it('update quick view agent quantity unknown', () => {
    expect(dosageReducer({}, updateQuickViewAgentQuantityUnknown(true))).toMatchSnapshot();
  });
});
