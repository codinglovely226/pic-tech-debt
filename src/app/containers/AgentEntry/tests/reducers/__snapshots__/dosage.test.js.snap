exports[`dosageReducer add agent 1`] = `
Object {
  "quantity": "1",
  "units": "mg",
  "unknown": false,
}
`;

exports[`dosageReducer returns the initial state 1`] = `
Object {
  "quantity": "",
  "units": "",
  "unknown": false,
}
`;

exports[`dosageReducer update agent quantity 1`] = `
Object {
  "quantity": "2",
  "units": "g",
  "unknown": false,
}
`;

exports[`dosageReducer update quick view agent quantity 1`] = `
Object {
  "quantity": 10,
  "units": "",
  "unknown": false,
}
`;

exports[`dosageReducer update quick view agent quantity units 1`] = `
Object {
  "quantity": "",
  "units": "kg",
  "unknown": false,
}
`;

exports[`dosageReducer update quick view agent quantity unknown 1`] = `
Object {
  "quantity": "",
  "units": "",
  "unknown": true,
}
`;
